permission_handler_platform_interface/lib/src/permission_handler_platform_interface.dart

```diff
@@ -64,7 +64,7 @@ abstract class PermissionHandlerPlatform extends PlatformInterface {
   ///
   /// Returns [true] if the app settings page could be opened, otherwise
   /// [false].
-  Future<bool> openAppSettings() {
+  Future<bool?> openAppSettings() {
```

Comment:
This method should never return `null` and should be enforced here.
```suggestion
  Future<bool> openAppSettings() {
```